solutions/fully-configurable/variables.tf

```diff
@@ -69,132 +73,118 @@ variable "service_plan" {
-# IAM engine config
+# KMS properties required when creating an encryption key, rather than passing an existing key CRN.
 ########################################################################################################################
 
-variable "iam_engine_enabled" {
+variable "kms_encryption_enabled" {
   type        = bool
-  description = "Set this to true to to configure a Secrets Manager IAM credentials engine. If set to false, no IAM engine will be configured for your instance."
+  description = "Set to true to enable Secrets Manager Secrets Encryption using customer managed keys. When set to true, a value must be passed for `existing_kms_instance_crn`. Cannot be set to true if passing a value for `existing_secrets_manager_crn` or `existing_secrets_manager_kms_key_crn`."
   default     = false
-}
 
-variable "iam_engine_name" {
-  type        = string
-  description = "The name of the IAM engine used to configure a Secrets Manager IAM credentials engine. If the prefix input variable is passed it is attached before the value in the format of '<prefix>-value'."
-  default     = "iam-engine"
-}
+  validation {
+    condition     = var.kms_encryption_enabled ? var.existing_secrets_manager_crn == null : true
+    error_message = "'kms_encryption_enabled' should be false if passing a value for 'existing_secrets_manager_crn'."
+  }
 
-########################################################################################################################
-# Key Protect
-########################################################################################################################
+  validation {
+    condition     = var.kms_encryption_enabled ? var.existing_secrets_manager_kms_key_crn == null : true
+    error_message = "'kms_encryption_enabled' should be false if passing a value for 'existing_secrets_manager_kms_key_crn'."
+  }
 
-variable "skip_kms_iam_authorization_policy" {
-  type        = bool
-  description = "Set to true to skip the creation of an IAM authorization policy that permits all Secrets Manager instances in the resource group to read the encryption key from the KMS instance. If set to false, pass in a value for the KMS instance in the `existing_kms_instance_crn` variable. If a value is specified for `ibmcloud_kms_api_key`, the policy is created in the KMS account."
-  default     = false
-}
+  validation {
+    condition     = var.existing_kms_instance_crn != null ? var.kms_encryption_enabled : true
+    error_message = "If passing a value for 'existing_kms_instance_crn', you should set 'kms_encryption_enabled' to true."
+  }
 
-variable "existing_secrets_manager_kms_key_crn" {
-  type        = string
-  description = "The CRN of a Key Protect or Hyper Protect Crypto Services key to use for Secrets Manager. If not specified, a key ring and key are created."
-  default     = null
+  validation {
+    condition     = var.kms_encryption_enabled ? (var.existing_kms_instance_crn != null ? true : false) : true
```

Comment:
this is not correct. The user could be passing a value for an existing key using `existing_secrets_manager_kms_key_crn`. Please check out the validation I used in https://github.com/terraform-ibm-modules/terraform-ibm-scc/blob/main/solutions/fully-configurable/variables.tf